pkg/apic/apiservicerevision.go

```diff
@@ -107,47 +107,54 @@ func (c *ServiceClient) GetAPIRevisions(queryParams map[string]string, stage str
 
 // getAPIRevisions - Returns the list of API revisions for the specified filter
 func (c *ServiceClient) getAPIRevisions(queryParams map[string]string, stage string) ([]v1alpha1.APIServiceRevision, error) {
-	headers, err := c.createHeader()
-	if err != nil {
-		return nil, err
-	}
+	apiRevisionsURL := c.cfg.GetRevisionsURL()
+	morePages := true
+	page := 1
 
-	request := coreapi.Request{
-		Method:      coreapi.GET,
-		URL:         c.cfg.GetRevisionsURL(),
-		Headers:     headers,
-		QueryParams: queryParams,
-	}
+	apiRevisions := make([]v1alpha1.APIServiceRevision, 0)
+	filteredAPIRevisions := make([]v1alpha1.APIServiceRevision, 0)
 
-	response, err := c.apiClient.Send(request)
-	if err != nil {
-		return nil, err
-	}
-	if response.Code != http.StatusOK {
-		if response.Code != http.StatusNotFound {
-			responseErr := readResponseErrors(response.Code, response.Body)
-			return nil, utilerrors.Wrap(ErrRequestQuery, responseErr)
+	for morePages {
```

Comment:
Should this be using the getResources func?

Also, can getResources be made public?  Then the pagination stuff in agent.go may utilize it as well?